### ISSUE TITLE
Cannot find any module matching the provided specification 'odoo'

### ISSUE THREAD

**Issue by max3903** [bitnami/bitnami-docker-odoo#87]
**Description**

I am trying to run the image on OKD 4.4.

**Steps to reproduce the issue:**

* I created the ImageStream and the BuildConfig to get a first version of the image in the registry
* I created the DeploymenConfig to run the image from the registry

**Describe the results you received:**

The pod failed with the following messages in the logs:

```
[0m
[0m[1mWelcome to the Bitnami odoo container[0m
[0mSubscribe to project updates by watching [1mhttps://github.com/bitnami/bitnami-docker-odoo[0m
[0mSubmit issues and feature requests at [1mhttps://github.com/bitnami/bitnami-docker-odoo/issues[0m
[0mSend us your feedback at [1mcontainers@bitnami.com[0m
[0m
[0m[38;5;2mINFO [0m ==> Starting odoo... 
[37mnami   [39m [31mERROR[39m Cannot find any module matching the provided specification 'odoo'
```
**Describe the results you expected:**

The pod is running and I can access Odoo web interface,

**Additional information you deem important (e.g. issue happens only occasionally):**

None

**Version**

```
$ oc version
Client Version: 4.4.0-0.okd-2020-05-23-055148-beta5
Server Version: 4.4.0-0.okd-2020-05-23-055148-beta5
Kubernetes Version: v1.17.1
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

Digital Ocean

**Comment by juan131** [comment]
Hi @max3903 

This image (Odoo) doesn't follow the "non-root" approach and it doesn't have support for arbitrary UIDs. Both of them are requirements for Openshfit.

While most of our catalog (specially infrastructure containers) follow this approach, some webapps (such as Odoo) don't follow it yet. Our plan is to migrate every webapp to this approach too (as we did for WordPress, see https://github.com/bitnami/bitnami-docker-wordpress#532-debian-10-r30) but it can take us some time until the full catalog of Bitnami images can be used on Openshift.

**Comment by max3903** [comment]
@juan131 Thanks! I guess I'll have to be patient then.

**Comment by firemound** [comment]
@max3903 you should be able to workaround the issue by creating a service account for any pods to run under and adding the anyuid policy scc to the service account. 

`oc adm policy add-scc-to-user anyuid -z odoo-sa`

**Comment by marcosbc** [comment]
Hi @max3903, just wanted to let you know that we released `12.0.20210615-debian-10-r20`/`13.0.20210610-debian-10-r24`/`14.0.20210610-debian-10-r22` versions of this container which includes a major refactor of the existing logic. Refer to the [Notable Changes section in the README](https://github.com/bitnami/bitnami-docker-odoo#notable-changes) for more info.

We tried to enable the non-root approach, but unfortunately it does not work because the Odoo application itself has limitations with respect to it, causing errors like these:

```
     File "/opt/bitnami/odoo/lib/odoo-14.0.post20210610-py3.7.egg/odoo/cli/server.py", line 40, in check_root_user
       if getpass.getuser() == 'root':
     File "/opt/bitnami/python/lib/python3.7/getpass.py", line 169, in getuser
       return pwd.getpwuid(os.getuid())[0]
   KeyError: 'getpwuid(): uid not found: 1001'
```

In the future, we may consider including [`nss_wrapper`](https://cwrap.org/nss_wrapper.html) to run as a non-root user, however that will require some changes to the container scripts. Therefore I'm going to leave this issue open as 'on-hold' (like before).

**Comment by carrodher** [comment]
Hi, unfortunately, if the issue is not solved in recent versions means the task is still on our backlog and there are no plans to work on it in the short/middle term.

In the same way, please, [note the Bitnami team is not actively working on new features/improvements nor providing support through GitHub Issues for this project](https://github.com/bitnami/bitnami-docker-odoo#community-supported-solution) since this asset was marked as a “Community supported solution”.

At this point, the best option to fix any issue or add a new feature is to contribute create a PR; the Bitnami team will be happy to review it and provide feedback.
